src/nyc_trees/js/src/lib/SelectableBlockfaceLayer.js

```diff
@@ -54,15 +54,21 @@ module.exports = BlockfaceLayer.extend({
     },
 
     addBlockface: function(data, latlng) {
-        if (data && data.geojson) {
-            var geom = mapUtil.parseGeoJSON(data.geojson);
-            if (this.options.onAdd(data, geom, latlng)) {
-                this.addData({
-                    "type": "Feature",
-                    "geometry": geom,
-                    "properties": data
-                });
-            }
+        var self = this;
+        if (!data || !data.id) {
+            return;
         }
+        mapUtil.fetchBlockface(data.id).then(function(blockfaceData) {
+            if (blockfaceData && blockfaceData.geojson) {
+                var geom = mapUtil.parseGeoJSON(blockfaceData.geojson);
+                if (self.options.onAdd(data, geom, latlng)) {
+                    self.addData({
+                        "type": "Feature",
+                        "geometry": geom,
+                        "properties": data
+                    });
+                }
+            }
+        });
```

Comment:
I'm seeing network requests when selecting blockfaces, even for ones we don't want to highlight (like clicking unavailable blocks on the reservations page).

Looking at the uses of `SelectableBlockfaceLayer.onAdd`, it looks like the only use of the `geom` parameter of `onAdd` is in [`surveyPage.js`](https://github.com/azavea/nyc-trees/blob/develop/src/nyc_trees/js/src/surveyPage.js#L133-L156).

I think we can modify that to move most of the logic into `onAdded` and drop the `geom` parameter from `onAdd` (which I should have probably named `shouldAdd`) by modifying `surveyPage.js` like so:
```javascript
    selectedLayer = new SelectableBlockfaceLayer(blockfaceMap, grid, {
        onAdd: function() {
            return true;
        },

        onAdded: function(feature, layer)
            var latLngs = mapUtil.getLatLngs(feature.geometry);

            blockfaceId = feature.properties.id;

            selectedLayer.clearLayers();
            endPointLayers.clearLayers();

            var startCircle = L.circleMarker(latLngs[0], defaultStyle),
                endCircle = L.circleMarker(latLngs[latLngs.length - 1], defaultStyle);

            startCircle.isStart = true;
            endCircle.isStart = false;

            endPointLayers.addLayer(startCircle);
            endPointLayers.addLayer(endCircle);

            showSelectStart();

            mapUtil.zoomToBlockface(blockfaceMap, blockfaceId);
        }
    }),
```

This will let us move the network request to *after* the call to `onAdd`, so we won't have to do it if `onAdd` returns false, like so:
```javascript
    addBlockface: function(data, latlng) {
        var self = this;
        if (!data || !data.id) {
            return;
        }
        if (self.options.onAdd(data, geom, latlng)) {
            mapUtil.fetchBlockface(data.id).then(function(blockfaceData) {
                if (blockfaceData && blockfaceData.geojson) {
                    var geom = mapUtil.parseGeoJSON(blockfaceData.geojson);
                    self.addData({
                        "type": "Feature",
                        "geometry": geom,
                        "properties": data
                    });
                }
            });
        }
    }
```